### ISSUE TITLE
Check uploaded video encoding status

### ISSUE THREAD

**Issue by josephharveyangeles** [vimeo/vimeo.js#89]
Is there any API endpoint to check if the video has finished both uploading and encoding? I'm uploading the video using the Node API, and then right after uploading I'm doing a client `GET` request on the just uploaded uri to fetch the thumbnails. It always fail, I'm getting an empty `body.data`. I'm assuming it was because the video is not available yet. (I'm on vimeo basic for now). But if I issue a separate `GET` request after a couple of seconds, everything's fine.

- Is the encoding process finished as well after uploading is done
- Will this be resolved when I upgrade my vimeo account to PRO and above?

**Comment by tommypenner** [comment]
Currently the API does not perform a callback or push notification when an upload has completed video transcoding. Instead, you'll need to periodically make requests to the API to get the video's status.

To get a video's status, make a [filtered request](https://developer.vimeo.com/api/common-formats#json-filter) with these fields specified:
```
GET https://api.vimeo.com/videos/[video_id]?fields=uri,upload.status,transcode.status
```
The `upload.status` and `transcode.status` fields will return one of the following values:
```
complete
error
in_progress
```

Generally, thumbnails aren't ready until the video has completed transcode. Upgrading to a paid account will put your videos higher up in the conversion queue, but otherwise has no effect on getting metadata from the API.

**Comment by josephharveyangeles** [comment]
Thanks for the reply @tommypenner, I think I'll just do that. 

I'm sorry, but one last thing. I can't seem to figure out how to set `privacy` stuff via API. I've checked the docs, but it doesn't really give all the possible values for the given parameters:

![image](https://user-images.githubusercontent.com/11130314/48340631-07864600-e6a6-11e8-8b7d-34cb58959d9a.png)

Where I can set the values for this parameters?
Basically, I want to the API equivalent of the following UI options:
- Who can watch? _Hide this vimeo from vimeo.com_
- Where can this be embedded? _Specific domains_ -> how can I specify via API?

**Comment by tommypenner** [comment]
@josephharveyangeles For `privacy.view` the accepted values and the view settings they correspond to are:
```
anybody => public
contacts => only people I follow
disable => hide from vimeo.com
nobody => only me
password => password
unlisted => private link
users => only people I choose
```

For `privacy.embed` the values are:
```
private => embedding is disabled
public => embed anywhere
whitelist => embed only on allowed domains
```

An overview of privacy settings can be found here: https://vimeo.zendesk.com/hc/en-us/articles/224817847-Privacy-settings-overview

**Comment by magbicaleman** [comment]
I just wanted to double-check on this. Through the Vimeo interface, when a file is transcoding we see the following information.

> Your video will be ready soon. Thanks for your patience.
> Approximate Time Remaining
> 00:01:18

Still as of now, there is no way to get that time. Is that still accurate?
